addons/mail/static/src/js/chat_manager.js

```diff
@@ -506,6 +507,8 @@ function on_toggle_star_notification (data) {
             chat_manager.bus.trigger('update_message', message);
         }
     });
+    starred_counter = _.where(messages, {is_starred: true}).length;
+    chat_manager.bus.trigger('update_starred', starred_counter);
```

Comment:
I don't understand the issue.  If a user clicks multiple times quickly on star button, then the chat manager will receive multiple notifications, and you can increment decrement appropriately...

Anyway, if it generates wrong values, then you need to understand and fix the reason why it generates those wrong values, not use an incorrect number to avoid the issue